### ISSUE TITLE
refactor the 'improved-sanity-test' to run on only HEAD

### ISSUE THREAD

**Issue by miabbott** [projectatomic/atomic-host-tests#150]
Our testing strategy with the `improved-sanity-test` has always had a known drawback:  if a problem exists in the HEAD-1 commit, the whole test fails and we miss out on testing the bits in the HEAD commit.

We've been able to live with this and workaround it via manual testing, but this is not going to scale.

It's time to re-think the test suite to test only the HEAD commit.  This will require the test suite to synthesize a commit to upgrade to and may not be a 'real-world' upgrade test, but I believe the trade-off is worth it for the coverage we gain.

**Comment by dustymabe** [comment]
does this mean we won't do an 'upgrade' or just that we won't run a ton of tests before the upgrade?

**Comment by mike-nguyen** [comment]
Off the top of my head I think @jlebon has steps for a simulated update so we can check for /etc /var 3-way merges, pkg layering through updates, etc.  Then maybe we can do what you do @dustymabe and upgrade to an older commit for RPM changes.  We're open to ideas because HEAD-1 is pretty confusing if you're looking at statuses and results.

**Comment by miabbott** [comment]
@jlebon showed us an example of this at one point.  It looks something like this (the `sed` line is probably not right):

```
# ostree refs --create local-branch <HEAD checksum>
# sed -i 's/^refspec=.*$/refspec=local-branch/' /path/to/origin/file/<HEAD checksum>.origin
# ostree commit -b local-branch --tree=ref=local-branch --add-metatdata-string version=<version>
# rpm-ostree upgrade
```

**Comment by miabbott** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [(5) Refactor the [a-h-t] sanity test to only require the HEAD commit to be deployed](https://trello.com/c/i2Dx4nFO/767-5-refactor-the-a-h-t-sanity-test-to-only-require-the-head-commit-to-be-deployed)

**Comment by dustymabe** [comment]
note: for my use case I wanted to be able to run tests on a disk image.. @miabbott already added the ability to do this in https://github.com/projectatomic/atomic-host-tests/commit/f15a61d02c4581be31b6a89bf31b2a64cf305558 with `--tag cloud_image`

**Comment by mike-nguyen** [comment]
#198 Will be merged on June 20th and resolve this issue

**Comment by miabbott** [comment]
Closed via #198
